src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/Dwarf/DwarfAbbrev.cs

```diff
@@ -0,0 +1,304 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using static ILCompiler.ObjectWriter.DwarfNative;
+
+namespace ILCompiler.ObjectWriter
+{
+    internal sealed class DwarfAbbrev
+    {
+        private readonly ushort[] _definition;
+
+        public static ushort DW_FORM_size = 0xDEAD; // Dummy value
+
+        public DwarfAbbrev(ushort[] definition)
+        {
+            _definition = definition;
+        }
+
+        public ushort Tag => _definition[0];
+
+        public bool HasChildren => _definition[1] == DW_CHILDREN_yes;
+
+        public int AbbreviationCode { get; set; }
```

Comment:
This is mutable, so statics below can change?

Reply:
Good point, that's a bad design. I'll move the abbreviation mapping to `DwarfInfoWriter` and remove this. Originally I used static abbreviation list and constant abbreviation codes but at some point I moved to tracking only the used ones and this got a bit messy.

Reply:
If AbbreviationCode  can be moved out without much trouble, DwarfAbbrev will be basically an array wrapper and could be `readonly struct`. 
Not that it will make a lot of difference for how this is used, just "why not" thing.

Reply:
Sounds reasonable.

Reply:
I didn't do the `readonly struct` change to keep the reference equality for `Dictionary`. On second though that's unwarranted concern, so I am keeping this "unresolved" to push the change with next batch.